sys/include/fs/littlefs_fs.h

```diff
@@ -72,6 +79,7 @@ typedef struct {
      * total number of block is defined in @p config.
      * if set to 0, the total number of sectors from the mtd is used */
     uint32_t base_addr;
+    uint16_t sectors_per_block; /**< number of sectors per block */
```

Comment:
This addition resulted in the subsequent buffers no longer being aligned two 4 bytes by accident, which triggered unaligned memory accesses. On Cortex M0+, such as used by the `samr21-xpro`, this then resulted in hard faults.

But the issue really is that alignment requirements need to be communicated to the compiler. It was just luck that this worked before.

Reply:
Well not so much luck since the `uint32_t` will always be word-aligned.

Moving `sectors_per_block` to end end of the struct should be the simple fix. (The buffers are already word-multiples)

Reply:
I doubt anyone placed them after a `uint32_t` member to align them without adding a comment to warn about breakage. To me, this looks like we goy lucky here :)

(E.g. `iolist_t` and `pkt_snip_t` have a common memory layout, but that is documented to avoid anyone from breaking it. I would expect similar comments here if that was inentionally.)